src/libraries/System.Text.Json/src/System/Text/Json/Reader/JsonReaderHelper.cs

```diff
@@ -383,5 +246,603 @@ public static bool TryGetFloatingPointConstant(ReadOnlySpan<byte> span, out doub
+                search = LoadVector(ref searchSpace, lengthToExamine);
+                offset = lengthToExamine;
+                search = Vector.BitwiseOr(
+                            Vector.BitwiseOr(
+                                Vector.Equals(search, values0),
+                                Vector.Equals(search, values1)),
+                            Vector.LessThan(search, valuesLessThan));
+                if (Vector<byte>.Zero.Equals(search))
+                {
+                    // None matched
+                    goto NotFound;
+                }
+
+            VectorMatch:
+                offset += (nuint)LocateFirstFoundByte(search);
+                goto Found;
+            }
+
+            Debug.Fail("Unreachable");
+            goto NotFound;
+        }
+#endif
+        // Vector sub-search adapted from https://github.com/aspnet/KestrelHttpServer/pull/1138
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static int LocateFirstFoundByte(Vector<byte> match)
+        {
+            var vector64 = Vector.AsVectorUInt64(match);
+            ulong candidate = 0;
+            int i = 0;
+            // Pattern unrolled by jit https://github.com/dotnet/coreclr/pull/8001
+            for (; i < Vector<ulong>.Count; i++)
+            {
+                candidate = vector64[i];
+                if (candidate != 0)
+                {
+                    break;
+                }
+            }
+
+            // Single LEA instruction with jitted const (using function result)
+            return i * 8 + LocateFirstFoundByte(candidate);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static int LocateFirstFoundByte(ulong match)
+        {
+            // Flag least significant power of two bit
+            var powerOfTwoFlag = match ^ (match - 1);
+            // Shift all powers of two into the high byte and extract
+            return (int)((powerOfTwoFlag * XorPowerOfTwoToHighByte) >> 57);
+        }
+
+        private const ulong XorPowerOfTwoToHighByte = (0x07ul |
+                                               0x06ul << 8 |
+                                               0x05ul << 16 |
+                                               0x04ul << 24 |
+                                               0x03ul << 32 |
+                                               0x02ul << 40 |
+                                               0x01ul << 48) + 1;
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static ref byte AddByteOffset(ref byte start, nuint offset)
+            => ref Unsafe.AddByteOffset(ref start, (IntPtr)(nint)offset);
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static Vector<byte> LoadVector(ref byte start, nuint offset)
+            => Unsafe.ReadUnaligned<Vector<byte>>(ref Unsafe.AddByteOffset(ref start, (IntPtr)(nint)offset));
```

Comment:
Can you please add a new file like "JsonReaderHelper.Intrinsics.cs" and move this code there.

Reply:
Done